src/commands/cmd_query.c

```diff
@@ -94,19 +94,17 @@ void _MGraph_Query(void *args) {
 	CommandCtx_ThreadSafeContextLock(qctx);
 	GraphContext *gc = GraphContext_Retrieve(ctx, qctx->graphName, readonly);
 	if(!gc) {
-		if(!AST_ContainsClause(ast, CYPHER_AST_CREATE) &&
-		   !AST_ContainsClause(ast, CYPHER_AST_MERGE)) {
-			CommandCtx_ThreadSafeContextUnlock(qctx);
-			RedisModule_ReplyWithError(ctx, "key doesn't contains a graph object.");
-			goto cleanup;
-		}
+
+		// Create.
 		gc = GraphContext_New(ctx, qctx->graphName, GRAPH_DEFAULT_NODE_CAP, GRAPH_DEFAULT_EDGE_CAP);
 
 		if(!gc) {
 			CommandCtx_ThreadSafeContextUnlock(qctx);
 			RedisModule_ReplyWithError(ctx, "Graph name already in use as a Redis key.");
 			goto cleanup;
 		}
+		// If in readonly mode, retrive key in read only privileges
+		if(readonly) gc = GraphContext_Retrieve(ctx, qctx->graphName, readonly);
```

Comment:
I don't _think_ this line is necessary, we can just use the `gc` retrieved at 99. The Redis key is closed in both `GraphContext_New` and `GraphContext_Retrieve`, I assume the read/write permissions are meaningless after key closure - correct me if I'm wrong.